### PR TITLE
Update code_checks.sh to check for instances of os.remove

### LINKED ISSUE #37095: Add CI check to check that `ensure_clean` is used instead of `os.remove`
xref https://github.com/pandas-dev/pandas/pull/34385#pullrequestreview-506705288

This could also be a pre-commit hook

Opening this issue as a placeholder so we don't forget

### PR DESCRIPTION
**Pull request by jnecus**
I have included a check for instances of os.remove in the tests folder. This is in line with issue #37095 whereby ensure_clean should be used rather than os.remove.

 A few outstanding questions remain:

- Should this be expanded to search for cases of os.remove elsewhere?

- ~~Cases of `os.remove ` still exist in common.py, test_writers.py, and test_store.py. For some reason, the case in test_store.py is not picked up by this check? Any idea why?~~ _(solved by passing arguments separately rather than as a list)_

- ~~Advice on how to implement this as a pre-commit hook also welcome.~~ _(to be implemented in separate PR)_

- [x] closes #37095 
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
`--exclude` doesn't take a list, I think you'll have to pass multiple arguments
```
 --exclude 'common.py' --exclude 'test_writers.py' 
```

**Comment by jnecus** [comment]
Perhaps it makes sense to filter the search for os.remove to specifically exclude the cases where it occurs currently, rather than exclude the whole files? If so any suggestions on preferred ways to achieve this?

File: ci/code_checks.sh [hunk 1]
```diff
@@ -292,6 +292,10 @@ if mods:
     "
     RET=$(($RET + $?)) ; echo $MSG "DONE"
 
+    MSG='Check code for instances of os.remove' ; echo $MSG
+    invgrep -R --include="*.py*" --exclude "common.py" --exclude "test_writers.py" --exclude "test_store.py" -E "os.remove" pandas/tests/
```
Comment (dsaxton) [inline ci/code_checks.sh:296]: Do we need to escape the `.` in `os.remove`?

Comment (jnecus) [reply ci/code_checks.sh:296]: I don't think so? `os.remove` is detected without escaping the `.` when files containing it are not excluded

Comment (MarcoGorelli) [reply ci/code_checks.sh:296]: OK but like this I think it'll also detect `oszremove`. You'll need to escape the `.` else it'll match anything

Comment (jnecus) [reply ci/code_checks.sh]: Sorry, got ya. Changed.

Comment (MarcoGorelli) [reply ci/code_checks.sh]: cool, thanks - I'm not sure about the location of the check though, as it's currently inside the "checks on imported code" section. Should it be in "looking for unwanted patterns"?

Comment (jnecus) [reply ci/code_checks.sh]: Makes sense, I've shifted it up to the 'patterns' section

**Review by MarcoGorelli — APPROVED** [review]
unless there's a better way of excluding exceptions than excluding entire files, this looks good to me :+1:

**Comment by jreback** [comment]
thanks @jnecus

**Pull request merged**